src/main/java/net/floodlightcontroller/statistics/web/BandwidthResource.java

```diff
@@ -66,7 +67,9 @@ public Object retrieve() {
                 spbs = new HashSet<SwitchPortBandwidth>(Collections.singleton(statisticsService.getBandwidthConsumption(dpid, port)));
             } else {
                 spbs = new HashSet<SwitchPortBandwidth>();
-                for (OFPortDesc pd : switchService.getSwitch(dpid).getPorts()) { /* do specific DPID; do all ports */
+                //fix concurrency scenario
+                IOFSwitch sw = switchService.getSwitch(dpid);
+                for (OFPortDesc pd : sw.getPorts()) { /* do specific DPID; do all ports */
```

Comment:
@xuraylei I would return a Map<String, String> with a single entry, ("error", "Switch X was not connected"):

{
    "error" : "Switch X was not connected"
}